tools/logger/convert.c

```diff
@@ -346,14 +359,20 @@ static void print_entry_params(FILE *out_fd,
 					   entry->header.component_class,
 					   dma_log->uid),
 			raw_output && strlen(ids) ? "-" : "",
-			ids,
-			to_usecs(dma_log->timestamp, clock),
-			dt,
-			format_file_name(entry->file_name, raw_output),
-			entry->header.line_idx);
+			ids);
+		fprintf(out_fd, time_fmt, to_usecs(dma_log->timestamp, clock),
+			dt);
+		if (!hide_location)
+			fprintf(out_fd, "(%s:%u) ",
+				format_file_name(entry->file_name, raw_output),
+				entry->header.line_idx);
 	} else {
 		/* timestamp */
-		fprintf(out_fd, "%s[%16.6f] (%16.6f)%s ",
+		snprintf(time_fmt, sizeof(time_fmt),
+			 "%%s[%%%d.%df] (%%%d.%df)%%s ",
+			 float_precision + 10, float_precision,
+			 float_precision + 10, float_precision);
```

Comment:
With a crazy long precision `snprintf` can truncate `time_fmt`. Reported by gcc 9.3 -Werr  and fixed in PR #3411